src/collection-preferences/content-display/index.tsx

```diff
@@ -130,37 +130,20 @@ export default function ContentDisplayPreference({
         role="list"
       >
         <DndContainer
-          sortedOptions={sortedAndFilteredOptions}
-          getId={option => option.id}
-          onChange={options => onChange(options.map(({ id, visible }) => ({ id, visible })))}
-          disabled={columnFilteringText.trim().length > 0}
-          renderOption={props => (
-            <li
-              className={clsx(
-                getOptionClassName(),
-                props.isDragging && styles.placeholder,
-                props.isSorting && styles.sorting
-              )}
-              style={props.style}
-            >
-              <ContentDisplayOption
-                ref={props.ref}
-                listeners={props.listeners}
-                dragHandleAriaLabel={props.dragHandleAriaLabel}
-                onToggle={onToggle}
-                option={props.option}
-                disabled={props.attributes['aria-disabled']}
-              />
-            </li>
-          )}
-          renderActiveOption={props => (
-            <ContentDisplayOption
-              option={props.option}
-              dragHandleAriaLabel={props.dragHandleAriaLabel}
-              listeners={props.listeners}
-              onToggle={onToggle}
-            />
-          )}
+          items={sortedAndFilteredOptions.map(data => ({ id: data.id, label: data.label, data }))}
```

Comment:
I changed option + getOptionId to items, where the ID and the label are provided as separate properties. That is similar to what we do in the Board component. An advantage is that item's metadata can be extended as necessary. The label is used to generate the ARIA labels for drag handles which is a combination of the i18n drag handle aria label and option label.